src/search/aggregations/bucket/bucket_selector_aggregation.rs

```diff
@@ -0,0 +1,182 @@
+use serde::{Serialize, Serializer};
+
+use crate::search::*;
+use crate::util::*;
+
+/// Specifies the path to the buckets to filter in a bucket selector aggregation.
+///
+/// This can either be a single path, referencing a single metric, or multiple paths
+/// in case of more complex aggregations.
+#[derive(Debug, Clone, PartialEq)]
+pub enum BucketsPath {
+    /// A single path referencing a metric.
+    Single(String),
+    /// Multiple paths in the form of key-value pairs.
+    /// Each key corresponds to an alias, and each value is a path to the metric.
+    Multi(Vec<(String, String)>),
```

Comment:
If these are key-value pairs, maybe it would make more sense to have it as
```suggestion
    Multi(Map<String, String>),
```

This way you could use `untagged` serde and discard custom serde implementation altogether.

For convenience, you could keep the regular From<(_, _)> implementations for easier DX.

Reply:
Fair point, I've updated it to use a map with untagged so no need for the custom serialisation